cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,19 +1,41 @@
+import {MenuContentPage, ProductListPage, LoginPage, ShopingCartPage,
+  AddressStepPage, ShippingStepPage, PaymentStepsPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productListPage = new ProductListPage();
+const loginPage = new LoginPage();
+const shopingCartPage = new ShopingCartPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepsPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin > span").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#cgv").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation > .button > span").click();
-    cy.get("#center_column > div > p > strong")
-        .should("have.text", "Your order on My Store is complete.");
+    // Visit Page
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    // Checkout process steps
+    productListPage.addToCart();
+    productListPage.goToCheckout();
+
+    // Shoping cart steps
+    shopingCartPage.goToCheckoutSumary();
+
+    // Login process steps
+    loginPage.login("aperdomobo@gmail.com", "WorkshopProtractor");
+    loginPage.submitLogin();
+
+    // Address confirmation steps
+    addressStepPage.goToCheckout();
+
+    // Shipping steps
+    shippingStepPage.acceptTemrs();
+    shippingStepPage.goToCheckoutShipping();
+
+    // Payment Steps
+    paymentStepPage.paymentMethods();
+    paymentStepPage.confirmFinalOrders();
+    paymentStepPage.orderConfirmations();
```

Comment:
The assertions should be done in the cy.js file

My suggestion is to retrieve the element on the Page Object and assert it on the test file